drivers/setup/blue/blue.c

```diff
@@ -835,29 +842,55 @@ ScrIoControl(
 {
     NTSTATUS Status;
     PIO_STACK_LOCATION stk = IoGetCurrentIrpStackLocation(Irp);
-    PDEVICE_EXTENSION DeviceExtension;
+    PDEVICE_EXTENSION DeviceExtension = DeviceObject->DeviceExtension;
 
-    DeviceExtension = DeviceObject->DeviceExtension;
     switch (stk->Parameters.DeviceIoControl.IoControlCode)
     {
         case IOCTL_CONSOLE_RESET_SCREEN:
         {
-            BOOLEAN Enable = !!*(PULONG)Irp->AssociatedIrp.SystemBuffer;
+            BOOLEAN Enable;
+
+            /* Validate input buffer */
+            if (Irp->AssociatedIrp.SystemBuffer == NULL)
+            {
+                Status = STATUS_INVALID_USER_BUFFER;
+                break;
+            }
+            if (stk->Parameters.DeviceIoControl.InputBufferLength < sizeof(ULONG))
+            {
+                Status = STATUS_INVALID_PARAMETER;
```

Comment:
Here it's more like "if the size is small your parameter is just fully incorrect" (since it contains a ULONG/BOOL that contains the state).
This `STATUS_INFO_LENGTH_MISMATCH` status seems to be better suitable for things like "query data related to some information class".